runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.service.ts

```diff
@@ -0,0 +1,100 @@
+import { TenantInviteCodeDto } from './dto/invite-code.dto';
+
+const relations = ['createdByUser'];
+
+@Injectable()
+export class TenantService {
+    private readonly logger = new Logger(TenantService.name);
+
+    constructor(
+        @InjectRepository(Tenant)
+        private readonly tenantRepository: Repository<Tenant>,
+        @InjectRepository(TenantInviteCode)
+        private readonly inviteCodeRepository: Repository<TenantInviteCode>
+    ) {}
+
+    getById(id: string) {
+        return this.tenantRepository.findOneBy({ id });
+    }
+
+    getAll() {
+        return this.tenantRepository.find({ relations })
+            .then((tenants) => tenants.map(tenant => ({
+                ...tenant,
+                createdByUser: {
+                    id: tenant.createdByUser.id,
+                    login: tenant.createdByUser.login
+                }
+            })));
+    }
+
+    async create(tenantDto: CreateTenantDto, requester: UserEntity): Promise<Tenant> {
+        const tenantByName = await this.tenantRepository.findOneBy({ name: tenantDto.name });
+        if (tenantByName) throw new BadRequestException('Name already exist', 'NameExist');
+
+
+        const newTenant = new Tenant();
+        newTenant.name = tenantDto.name;
+        newTenant.createdBy = requester.id;
+        newTenant.lastModifiedBy = requester.email;
+
+        return this.tenantRepository.save(newTenant);
+    }
+
+    async update(
+        tenantDto: UpdateTenantDto,
+        id: string,
+        requester: UserEntity
+    ): Promise<Tenant> {
+        const tenantByName = await this.tenantRepository.findOneBy({ name: tenantDto.name });
+        if (tenantByName) throw new BadRequestException('Name already exist', 'NameExist');
+
+        const updatedTenant = await this.tenantRepository
+            .findOneByOrFail({ id })
+            .then(tenant => ({
+                ...tenant,
+                ...tenantDto,
+                lastModifiedBy: requester.email
+            })).catch(() => {
+                this.logger.error('Cannot find tenant with id: ', id);
+                throw new BadRequestException('Tenant not found', 'NotFound');
+            });
+
+        return this.tenantRepository.save(updatedTenant);
+    }
+
+    getActiveInviteCodeByTenantId(tenantId: string): Promise<TenantInviteCodeDto | null> {
+        return this.inviteCodeRepository.findOneBy({
+            tenantId,
+            expirationDate: MoreThanOrEqual(new Date())
+        }).then(inviteCode => inviteCode ? ({ inviteCode: inviteCode.inviteId }) : null);
+    }
+
+    async createInviteCodeByTenantId(tenantId: string) {
+        const tenant = await this.tenantRepository
+            .findOneByOrFail({ id: tenantId })
+            .catch(() => {
+                throw new NotFoundException('Cannot find tenant with id: ', tenantId);
+            });
+
+        const newInviteCode = {
+            tenantId: tenant.id,
+            creationDate: dayjs(),
```

Comment:
we shouldn't persist non utc time, it's better to save timestamps directly in database

Reply:
I checked it and I suppose that because of entity definition - time is being saved in UTC